packages/datadog-instrumentations/src/fastify.js

```diff
@@ -45,16 +46,35 @@ function wrapAddHook (addHook) {
 
     if (typeof fn !== 'function') return addHook.apply(this, arguments)
 
-    arguments[arguments.length - 1] = shimmer.wrapFunction(fn, fn => function (request, reply, done) {
+    arguments[arguments.length - 1] = shimmer.wrapFunction(fn, fn => function () {
+      const request = arguments[0]
+      const reply = arguments[1]
       const req = getReq(request)
 
       try {
-        if (typeof done === 'function') {
-          done = arguments[arguments.length - 1]
+        // done callback is always the last argument
+        const done = arguments[arguments.length - 1]
 
+        if (typeof done === 'function') {
           arguments[arguments.length - 1] = function (err) {
             publishError(err, req)
 
+            const hasCookies = request.cookies && Object.keys(request.cookies).length > 0
```

Comment:
eeeeh, usually we check that in the listener instead, not sure it makes a big difference, but when debugging it's nice to still have the DC event fired, even if it is with empty object

Reply:
I talked with Ruben in slack about this last time, and from what I understand, it's really not THAT expensive, it's just a new object, nothing crazy